SampleAPI/Repository/EmployeeRepository.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SampleAPI.Constant;
+using SampleAPI.Contexts;
+using SampleAPI.Models;
+
+namespace SampleAPI.Repository
+{
+    /// <summary>
+    /// EmployeeRepository is used to perform CRUD operations with employee entity.
+    /// </summary>
+    public class EmployeeRepository : IEmployeeRepository
+    {
+        private readonly EmployeeContext _dbContext;
+
+        public EmployeeRepository(EmployeeContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        /// <summary>
+        /// Add new employee record to the database.
+        /// </summary>
+        /// <param name="employeeModel">Employee model</param>
+        public async Task AddAsync(EmployeeModel employeeModel)
+        {
+            if (employeeModel != null)
+            {
+                _dbContext.Employees.Add(employeeModel);
+                await _dbContext.SaveChangesAsync();
+            }
+        }
+
+        /// <summary>
+        /// Delete employee from the database.
+        /// </summary>
+        /// <param name="employeeId">Employee model.</param>
+        public async Task DeleteAsync(int employeeId)
+        {
+            var employee = await _dbContext.Employees.FindAsync(employeeId);
+
+            if (employee == null)
+            {
+                throw new CustomException(AppConstants.NoRecordErrorMessage);
+            }
+
+            _dbContext.Employees.Remove(employee);
+            await _dbContext.SaveChangesAsync();
+        }
+
+        /// <summary>
+        /// Get the employee list.
+        /// </summary>
+        /// <returns>Employee list.</returns>
+        public async Task<IEnumerable<EmployeeModel>> GetAllAsync(string searchText, string role, bool includeInActive)
+        {
+             var employeeList = await _dbContext.Employees.ToListAsync();
+            if (!string.IsNullOrWhiteSpace(searchText))
+            {
+                employeeList = employeeList.Where(x => x.Name.Contains(searchText) || x.Role.Contains(searchText)).ToList();
```

Comment:
doing a ToList() on each Where clause is not performance wise acceptable. You are iterating through the collection everytime that you have a where clause. Instead you should add all of the where clause at once and then iterate over the collection just once